transcom/mymove
# B 17906 Customer Request after SIT Expiration

## [B-17906](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A867878)

## Summary

This story prevents Prime from updating the dates for Authorized End Date if the new calculation would be after the current Authorized End Date. Customer dates should not extend the Authorized End Date beyond its current date. This avoids an automatic SIT Extension. 

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Functionality Test

1. Create a new HHG shipment in the Customer app and submit it
2. Login as a Service Counselor and approve the shipment
3. Login as a TOO and accept the shipment
4. Login as a Prime into the Prime Simulator
5. Select the newly created move
6. Update the shipment and make sure you enter values for the estimated and actual weights, this is needed to calculate the required delivery date. Note the weight entered.
7. Click Add Service Item
8. Add a new Origin SIT item, enter today's date for SIT Entry Date
9. Login as a TOO access the move and go to Move Task Order tab.
10. Accept the Domestic origin 1st day SIT Service Item
11. Login as a Prime to the Prime Simulator
12. Select the Move
13. Click on Edit for DOFSIT - Domestic origin 1st day SIT
14. Enter SIT Departure Date
15. For SIT Requested Delivery and SIT Customer Contacted dates enter dates that are more than 2 months in the future.
16. Click Save
17. You should see an error "dates entered cannot extend authorized end date beyond its current date"

Authorized End Date for Origin SIT: sitAuthorizedEndDate should be GracePeriodDays days after sitCustomerContacted or the sitDepartureDate whichever is earlier.

### Backend
